gym_electric_motor/physical_systems/electric_motors/externally_excited_synchronous_motor.py

```diff
@@ -167,40 +177,39 @@ def _torque_limit(self):
             _q = - i_n ** 2 / 2
             if mp['l_d'] < mp['l_q']:
-                i_d_opt = - _p / 2 - np.sqrt( (_p / 2) ** 2 - _q)
+                i_d_opt = - _p / 2 - np.sqrt((_p / 2) ** 2 - _q)
             else:
-                i_d_opt = - _p / 2 + np.sqrt( (_p / 2) ** 2 - _q)
+                i_d_opt = - _p / 2 + np.sqrt((_p / 2) ** 2 - _q)
             i_q_opt = np.sqrt(i_n ** 2 - i_d_opt ** 2)
         return self.torque([i_d_opt, i_q_opt, self._limits['i_e'], 0])
 
     def torque(self, currents):
         # Docstring of superclass
         mp = self._motor_parameter
-        return 1.5 * mp['p'] * (mp['l_m'] * currents[self.I_E_IDX] + (mp['l_d'] - mp['l_q']) * currents[self.I_SD_IDX]) * currents[self.I_SQ_IDX]
+        return 1.5 * mp['p'] * (mp['l_M'] * currents[self.I_E_IDX] * mp['i_k_rs'] + (mp['l_d'] - mp['l_q']) * currents[self.I_SD_IDX]) * currents[self.I_SQ_IDX]
 
     def electrical_jacobian(self, state, u_in, omega, *args):
         mp = self._motor_parameter
-        sigma = 1 - mp['l_m'] ** 2 / (mp['l_d'] * mp['l_e'])
         return (
             np.array([ # dx'/dx
-                [                                 -mp['r_s'] / mp['l_d'],                          mp['l_q'] / (sigma * mp['l_d']) * omega * mp['p'], mp['l_m'] * mp['r_e'] / (sigma * mp['l_d'] * mp['l_e']), 0],
-                [               -mp['l_d'] / mp['l_q'] * omega * mp['p'],                                                     -mp['r_s'] / mp['l_q'],                -omega * mp['p'] * mp['l_e'] / mp['l_q'], 0],
-                [mp['l_m'] * mp['r_s'] / (sigma * mp['l_d'] * mp['l_e']), -omega * mp['p'] * mp['l_m'] * mp['l_q'] / (sigma * mp['l_d'] * mp['l_e']),                                  -mp['r_e'] / mp['l_e'], 0],
-                [                                                      0,                                                                          0,                                                       0, 0],
+                [                       -mp['r_s'] / (mp['l_d'] * mp['sigma']),                          mp['l_q'] / (mp['sigma'] * mp['l_d']) * omega * mp['p'], mp['l_M'] * mp['r_e'] / (mp['sigma'] * mp['l_d'] * mp['l_E']) * mp['i_k_rs'], 0],
+                [                     -mp['l_d'] / mp['l_q'] * omega * mp['p'],                                                           -mp['r_s'] / mp['l_q'],                      -omega * mp['p'] * mp['l_E'] / mp['l_q'] * mp['i_k_rs'], 0],
+                [mp['l_M'] * mp['r_s'] / (mp['sigma'] * mp['l_d'] * mp['l_E']), -omega * mp['p'] * mp['l_M'] * mp['l_q'] / (mp['sigma'] * mp['l_d'] * mp['l_E']),                                        -mp['r_E'] / mp['l_E'] * mp['i_k_rs'], 0],
+                [                                                            0,                                                                                0,                                                                            0, 0],
             ]),
```

Comment:
I believe lines 198-201 should be replaced with 
                [                                     -mp['r_s'] / (mp['l_d'] * mp['sigma']),                                        mp['l_q'] / (mp['sigma'] * mp['l_d']) * omega * mp['p'], mp['l_M'] * mp['r_E'] / (mp['sigma'] * mp['l_d'] * mp['l_E']) * mp['i_k_rs'], 0],
                [                                   -mp['l_d'] / mp['l_q'] * omega * mp['p'],                                                                         -mp['r_s'] / mp['l_q'],                      -omega * mp['p'] * mp['l_M'] / mp['l_q'] * mp['i_k_rs'], 0],
                [mp['l_M'] * mp['r_s'] / (mp['sigma'] * mp['l_d'] * mp['l_E']* mp['i_k_rs']), -omega * mp['p'] * mp['l_M'] * mp['l_q'] / (mp['sigma'] * mp['l_d'] * mp['l_E']* mp['i_k_rs']),                                      -mp['r_E'] / (mp['sigma'] * mp['l_E'])], 0],
                [                                                                          0,                                                                                0,                                                                                          0, 0],

Reply:
I have the same solution.